harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -751,14 +751,16 @@ def _train_for_op(
         self, op: core.SearcherOperation, train_boundaries: List[_TrainBoundary]
     ) -> None:
         if self.local_training:
-            searcher_length = self.max_length
+            train_length = self.max_length
         else:
-            searcher_length = TrainUnit._from_searcher_unit(
+            train_length = TrainUnit._from_searcher_unit(
                 op.length, self.searcher_unit, self.global_batch_size
             )
-        assert searcher_length
+        if self.test_mode:
+            train_length = Batch(1)
```

Comment:
nit: reorder your if statements so you aren't mutating variables; that would feel a lot more natural:
```python
if self.test_mode:
  train_length = Batch(1)
elif self.local_training:
  train_length = self.max_length
else:
  train_length = TrainUnit._from_searcher_unit(...)
```

Reply:
duh. thanks.